lib/project/cordova-project.ts

```diff
@@ -0,0 +1,195 @@
+		private $templatesService: ITemplatesService,
+		$resources: IResourceLoader,
+		private $serverExtensionsService: IServerExtensionsService,
+		private $server: Server.IServer) {
+		super($logger, $fs, $resources);
+	}
+
+	public get name(): string {
+		return this.$projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.Cordova;
+	}
+
+	public get capabilities(): IProjectCapabilities {
+		return {
+			build: true,
+			buildCompanion: true,
+			deploy: true,
+			simulate: true,
+			livesync: true,
+			livesyncCompanion: true,
+			updateKendo: true
+		};
+	}
+
+	public get defaultProjectTemplate(): string {
+		return this.$config.DEFAULT_CORDOVA_PROJECT_TEMPLATE;
+	}
+
+	public get liveSyncUrl(): string {
+		return "icenium://";
+	}
+
+	public get requiredAndroidApiLevel(): number {
+		return 10;  // 2.3 Gingerbread
+	}
+
+	public get configFiles(): Project.IConfigurationFile[] {
+		var allConfigFiles = this.$projectFilesManager.availableConfigFiles;
+		return _.values(allConfigFiles);
+	}
+
+	public getProjectTargets(projectDir: string): IFuture<string[]> {
+		var fileMask = /^cordova\.(\w*)\.js$/i;
+		return this.getProjectTargetsBase(projectDir, fileMask);
+	}
+
+	public getTemplateFilename(name: string): string {
+		return util.format("Telerik.Mobile.Cordova.%s.zip", name);
+	}
+
+	public alterPropertiesForNewProject(properties: any, projectName: string): void {
+		this.alterPropertiesForNewProjectBase(properties, projectName);
+	}
+
+	public projectTemplatesString(): IFuture<string> {
+		return this.$templatesService.getTemplatesString(/.*Telerik\.Mobile\.Cordova\.(.+)\.zip/);
+	}
+
+	public getProjectFileSchema(): IFuture<any> {
+		return this.getProjectFileSchemaByName(this.name);
+	}
+
+	public getFullProjectFileSchema(): IFuture<any> {
+		return this.getFullProjectFileSchemaByName(this.name);
+	}
+
+	public adjustBuildProperties(buildProperties: any, projectData?: IProjectData): any {
+		buildProperties.CorePlugins = projectData.CorePlugins;
+		return buildProperties;
+	}
+
+	public ensureAllPlatformAssets(projectDir: string, frameworkVersion: string): IFuture<void> {
+		return (() => {
+			var platforms = _.keys(MobileHelper.platformCapabilities);
+			_.each(platforms, (platform: string) => this.ensureCordovaJs(platform, projectDir, frameworkVersion).wait());
+		}).future<void>()();
+	}
+
+	private ensureCordovaJs(platform: string, projectDir: string, frameworkVersion: string): IFuture<void> {
+		return (() => {
+			var cordovaJsFileName = path.join(projectDir, util.format("cordova.%s.js", platform).toLowerCase());
+			if (!this.$fs.exists(cordovaJsFileName).wait()) {
+				this.printAssetUpdateMessage();
+				var cordovaJsSourceFilePath = this.$resources.buildCordovaJsFilePath(frameworkVersion, platform);
+				this.$fs.copyFile(cordovaJsSourceFilePath, cordovaJsFileName).wait();
+			}
+		}).future<void>()();
+	}
+
+	public getSimulatorParams(projectDir: string, projectData: IProjectData, simulatorPackageName: string): IFuture<string[]> {
```

Comment:
I am not entirely sure that the project should know how to launch the simulator. I believe these functions belong to the simulator class.